javascript/ql/src/Security/CWE-400/PrototypePollutionUtility.ql

```diff
@@ -168,6 +241,7 @@ predicate dynamicPropWrite(DataFlow::Node base, DataFlow::Node prop, DataFlow::N
   exists(AssignExpr write, IndexExpr index |
     index = write.getLhs() and
     base = index.getBase().flow() and
+    isPotentiallyObjectPrototype(base.getALocalSource()) and
```

Comment:
Can we add a comment that this is mostly a heuristic pruning step, and that the full query ideally didn't need this early check. 
I get that we also get rid of a FP with this change, but I believe we ideally would get rid of that FP without a pre-analysis that is artificially limited by for instance not tracking through (non-existent) returns.